src/Umbraco.Web.UI.Client/src/views/common/infiniteeditors/linkpicker/linkpicker.controller.js

```diff
@@ -64,8 +64,21 @@ angular.module("umbraco").controller("Umbraco.Editors.LinkPickerController",
                 //will be either a udi or an int
                 var id = $scope.model.target.udi ? $scope.model.target.udi : $scope.model.target.id;
 
-                // is it a content link?
-                if (!$scope.model.target.isMedia) {
+                if ($scope.model.target.udi) {
+                    // extract the entity type from the udi and set target.isMedia accordingly
+                    var entityType = id.substring("umb://".length, id.lastIndexOf("/"));
```

Comment:
I feel like we should have some sort of utility class for doing things like this in js like `udiParser` or something then we can add tests for it

Reply:
Makes sence, I've added a `udiHelper` and some tests

Reply:
We are trying to stop using the term `helper` for any new classes/objects/etc... maybe rename to something like `udiParser` ?

Reply:
I've renamed to it `udiParser`